w09h01/test/pgdp/pingutrip/LegalCheck.java

```diff
@@ -0,0 +1,260 @@
+// ************* PLEASE READ THIS DISCLAIMER ************* //
+/*                                                          *
+ * Even when there is a test provided by Artemis to test    *
+ * the usage of not allowed things, you could still lose    *
+ * points after the deadline, if you are using not allowed  *
+ * things. See the link below                               *
```

Comment:
Currently, the Artemis test is better (it can detect everything that this test can detect and can detect other things like ternary) than our test but maybe this will change it the future

Reply:
Our test can detect ternary!

Reply:
When I'm using `return wayPoints == null ? a : b` the test still passes 👀

Reply:
That’s a bug then, not a lack of feature. You can add the question mark to the banned phrases and rewrite the symbol part of the lexer to create nodes of words instead i suppose